azimuth/utils/exclude_fields_with_extra.py

```diff
@@ -0,0 +1,40 @@
+# Copyright ServiceNow, Inc. 2021 – 2022
+# This source code is licensed under the Apache 2.0 license found in the LICENSE file
+# in the root directory of this source tree.
+
+from typing import Collection, Dict, Mapping, TypeVar
+
+from pydantic import BaseModel
+
+Model = TypeVar("Model", bound=BaseModel)
+
+
+def exclude_fields_with_extra(model: Model, extra: str):
```

Comment:
We can revisit how "specific" we want this function to be. Right now, it accepts a parameter `extra` and we pass it `"exclude_from_cache"` in 2 places (plus 1 in unit tests). That's useful if one day we have a use case for it that is other than `"exclude_from_cache"`.
Alternatively, we could remove the `extra` parameter and hardcode `"exclude_from_cache"` inside, so we don't have to pass it in 2 places. We should then rename the function and the file to be that specific.
```suggestion
def exclude_fields_from_cache(model: Model):
```
In between, we can give `extra` a default value of `"exclude_from_cache"`.
```suggestion
def exclude_fields_with_extra(model: Model, extra: str = "exclude_from_cache"):
```
What do you think?

Reply:
I'm ok with either - what would be the best practice? Since we only have one use case for now, YAGNI?

Reply:
Oh-ho! So extreme programming terms! 🤟 

Let's think about it...

The `extra` argument is not really an "extra functionality". For example, It's not adding unnecessary lines of code that we would then have to maintain unnecessarily, so it doesn't hurt.

I think the main difference between those is the name of the function.
```py
exclude_fields_with_extra(self.config, "exclude_from_cache")
```
Making the function generic like that allowed me to choose a name I find explains very well what the function does, then the argument like `"exclude_from_cache"` makes it clear for the caller what fields will be excluded. We can see that like separation of concerns; the functionality/logic vs what fields need to be excluded.

I am not able to convey those two pieces of information in one function name, we can't tell what starts where:
```py
exclude_fields_with_exclude_from_cache(self.config)
```
But it might not be necessary. 🤷‍♂️ Maybe the following is clear enough, and the slight benefit of not duplicating the string `"exclude_from_cache"` in two places is worth it.
```py
exclude_fields_from_cache(self.config)
```

Reply:
I think I'll go with
```py
exclude_fields_from_cache(self.config)
```
The function's name gives less detail about how the function works, but I have the opportunity to make the docstring more straightforward and clear:
```diff
-    """Exclude pydantic fields with given extra keyword argument.
+    """Exclude pydantic fields with exclude_from_cache=True.
```

Reply:
I'll let you decide, I think that both are clear.